File: Striver_SDE_SHEET/55-jump-game/jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size();
        int curr = 0;
        for(int i=0;i<n;i++){
            if(i>curr) return false;
            curr = max(curr,nums[i]+i);
        }
        return true;
    }
};

File: Striver_SDE_SHEET/55-jump-game/jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        max_i = 0
        for i in range(n):
            if i>max_i: return False
            max_i = max(max_i,i+nums[i])
        return True
                


                